packages/tldraw/src/lib/tools/SelectTool/childStates/EditingShape.ts

```diff
@@ -116,7 +116,7 @@ export class EditingShape extends StateNode {
 				// N.B. One nuance here is that we want empty text fields to be removed from the canvas when the user clicks away from them.
 				const isEmptyTextShape =
 					this.editor.isShapeOfType<TLTextShape>(editingShape, 'text') &&
-					renderPlaintextFromRichText(this.editor, editingShape.props.richText).trim() === ''
+					isEmptyRichText(editingShape.props.richText)
```

Comment:
this isn't quite the same - we want the shapes to be removed if they're just whitespace, going to revert this bit

Reply:
Good catch, I'd left the one in onShapeUpdate or EditEnd or whatever that deletes the shape, though with a note that we should see if we can do a version of tiptap/pm's `getText` that bails as soon as it finds any text, so that we don't block after editing large pieces of text.